bin/prune.sh

```diff
@@ -17,6 +19,18 @@ PAST=`create_past_yyyymmdd ${DELETE_TARGET_DAYS_LEFT}`
 # if it exists, delete it
 TARBALL_PAST="${BACKUPFILE_PREFIX}-${PAST}.tar.bz2"
 
+TARGET_BUCKET_URL=${TARGET_PRIVATE_BUCKET_URL}
+
+echo "pruning ${TARGET_BUCKET_URL}"
+if [ `echo $TARGET_BUCKET_URL | cut -f1 -d":"` == "s3" ]; then
+  s3_delete_file_if_delete_backup_day ${TARGET_BUCKET_URL}${TARBALL_PAST} ${DELETE_TARGET_DAYS_LEFT} ${DELETE_DEVIDE}
+elif [ `echo $TARGET_BUCKET_URL | cut -f1 -d":"` == "gs" ]; then
+  gs_delete_file_if_delete_backup_day ${TARGET_BUCKET_URL}${TARBALL_PAST} ${DELETE_TARGET_DAYS_LEFT} ${DELETE_DEVIDE}
+fi
+
+TARGET_BUCKET_URL=${TARGET_PUBLIC_BUCKET_URL}
```

Comment:
The pruning logic looks identical (unless I'm missing something), so rather than re-binding `TARGET_PUBLIC_URL`, would you consider a `for`?

```bash
for TARGET_BUCKET_UL in ${TARGET_PRIVATE_BUCKET_URL} ${TARGET_PUBLIC_BUCKET_URL}
```

Reply:
(Here and also in `list.sh`)

Reply:
You are right, I will add the prune and list support for R2 as well

Reply:
Done!

Reply:
In my current diff, lines 30-37 and 39-46 are identical logic. Similar in `list.sh`. Would you please implement the `for` loop iteration, or suggest why not to?

e.g.

```bash
for TARGET_BUCKET_URL in ${TARGET_PRIVATE_BUCKET_URL} ${TARGET_PUBLIC_BUCKET_URL}
do
  echo "pruning ${TARGET_BUCKET_URL}"
  if [ `echo $TARGET_BUCKET_URL | cut -f1 -d":"` == "s3" ]; then
    s3_delete_file_if_delete_backup_day ${TARGET_BUCKET_URL}${TARBALL_PAST} ${DELETE_TARGET_DAYS_LEFT} ${DELETE_DEVIDE}
  elif [ `echo $TARGET_BUCKET_URL | cut -f1 -d":"` == "gs" ]; then
    gs_delete_file_if_delete_backup_day ${TARGET_BUCKET_URL}${TARBALL_PAST} ${DELETE_TARGET_DAYS_LEFT} ${DELETE_DEVIDE}
  fi
done
```

Reply:
Done!